modules/code_metrics.js

```diff
@@ -10,11 +10,15 @@ function analyze(code) {
   const result = escomplex.analyse(code);
 
   // Round to 3 decimal places if needed
+  roundedDensity = +(result.aggregate.cyclomaticDensity.toFixed(3));
```

Comment:
in `routes/problem_id.js`, i think we should add a line that stores density as a datastore double so that the ranking is correct